iothub/device/src/ClientFactory.cs

```diff
@@ -83,11 +83,6 @@ public static InternalClient Create(string hostname, string gatewayHostname, IAu
                     throw new ArgumentException("certificate must be present in DeviceAuthenticationWithX509Certificate");
                 }
 
-                if (!connectionStringBuilder.Certificate.HasPrivateKey)
```

Comment:
The availability of the client certificate's private key will be validated when server presents the crypto challenge to the client.

Reply:
Should these if blocks be preserved when the user isn't on Netstandard2.1? From what I can tell, the remote callback only gets executed for Netstandard2.1

Reply:
This check is independent of the remote certificate validation callback. This is the client certificate that is presented to the remote server. The service doesn't actually require the private key, but instead relies on the client solving the challenge, in order to establish the mutual TLS connection.